### PR TITLE
Add schema link and signature

### PR DESCRIPTION
**Pull request by jefer94**

### CODE REVIEW

File: breathecode/assignments/tasks.py [hunk 1]
```diff
@@ -115,4 +116,21 @@ def serialize_task(task):
     cohort_user.history_log = user_history_log
     cohort_user.save()
 
+    s = None
```
Comment (jefer94) [inline breathecode/assignments/tasks.py:119]: Test

File: breathecode/assignments/views.py [hunk 2]
```diff
@@ -818,3 +819,22 @@ def put(self, request, task_id):
         item.save()
 
         return Response(item.subtasks)
+
+
+class MeCodeRevisionView(APIView):
+
+    def get(self, request):
```
Comment (jefer94) [inline breathecode/assignments/views.py:826]: Test

File: breathecode/assignments/views.py [hunk 3]
```diff
@@ -818,3 +819,22 @@ def put(self, request, task_id):
...
+        s = Service('rigobot', request.user.id)
+        response = s.get('/v1/finetuning/me/coderevision?read=false')
+        return response.json()
+
+
+class MeTaskCodeRevisionView(APIView):
+
+    def get(self, request, task_id):
```
Comment (jefer94) [inline breathecode/assignments/views.py:834]: Test

File: breathecode/authenticate/management/commands/sign_jwt.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import os
+from django.core.management.base import BaseCommand
+
+HOST = os.environ.get('OLD_BREATHECODE_API')
+DATETIME_FORMAT = '%Y-%m-%d'
+
+
+class Command(BaseCommand):
```
Comment (jefer94) [inline breathecode/authenticate/management/commands/sign_jwt.py:8]: Test

File: breathecode/authenticate/management/commands/sign_request.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import os
+from django.core.management.base import BaseCommand
+
+HOST = os.environ.get('OLD_BREATHECODE_API')
+DATETIME_FORMAT = '%Y-%m-%d'
+
+
+class Command(BaseCommand):
```
Comment (jefer94) [inline breathecode/authenticate/management/commands/sign_request.py:8]: Test

File: breathecode/authenticate/models.py [hunk 6]
```diff
@@ -111,6 +113,242 @@ def __str__(self):
         return f'{self.name} ({self.slug})'
 
 
+class Scope(models.Model):
```
Comment (jefer94) [inline breathecode/authenticate/models.py:116]: Change the models in rigobot

File: breathecode/authenticate/models.py [hunk 7]
```diff
@@ -111,6 +113,242 @@ def __str__(self):
...
+        self.__class__.objects.exclude(app_user_agreement__id__gte=1).exclude(id=self.id).delete()
+
+        if had_pk:
+            reset_app_user_cache()
+
+
+class AppUserAgreement(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    app = models.ForeignKey(App, on_delete=models.CASCADE)
+    optional_scope_set = models.ForeignKey(OptionalScopeSet,
```
Comment (jefer94) [inline breathecode/authenticate/models.py:294]: Do an endpoint that check if the user accept the last version of the agreement

File: breathecode/authenticate/tasks.py [hunk 8]
```diff
@@ -73,3 +74,9 @@ def async_accept_user_from_waiting_list(user_invite_id: int) -> None:
             'SUBJECT': 'Set your password at 4Geeks',
             'LINK': os.getenv('API_URL', '') + f'/v1/auth/password/{invite.token}'
         })
+
+
+@shared_task
+def destroy_legacy_key(legacy_key_id):
+    from .models import LegacyKey
+    LegacyKey.objects.filter(id=legacy_key_id).delete()
```
Comment (jefer94) [inline breathecode/authenticate/tasks.py:82]: Reset cache

File: breathecode/authenticate/views.py [hunk 9]
```diff
@@ -2287,3 +2289,146 @@ def delete(self, request):
...
+    ...
+
+    # send the webhook to celery
+    ...
+
+    return Response({'message': 'ok'})
+
+
+# demo
+class DemoView(APIView):
```
Comment (jefer94) [inline breathecode/authenticate/views.py:2358]: Remove it

File: breathecode/authenticate/views.py [hunk 10]
```diff
@@ -2287,3 +2289,146 @@ def delete(self, request):
...
+        # force signature
+        s = Service('rigobot', request.user.id, mode='signature')
+        request = s.get('https://rigobot.4geeks.com/webhook')
+        response = request.json()
+
+        return Response({'message': 'ok'})
+
+
+@private_view()
+def authorize_view(request, token=None, app_slug=None):
```
Comment (jefer94) [inline breathecode/authenticate/views.py:2384]: Test

File: breathecode/authenticate/views.py [hunk 11]
```diff
@@ -2287,3 +2289,146 @@ def delete(self, request):
         instance.delete()
 
         return Response(None, status=status.HTTP_204_NO_CONTENT)
+
+
+# app/user/:id
+class AppUserView(APIView):
```
Comment (jefer94) [inline breathecode/authenticate/views.py:2295]: Test

File: breathecode/authenticate/views.py [hunk 12]
```diff
@@ -2287,3 +2289,146 @@ def delete(self, request):
...
+        serializer = AppUserSerializer(items, many=True)
+
+        return handler.response(serializer.data)
+
+
+# app/webhook
+@api_view(['POST'])
+@permission_classes([AllowAny])
+@scope(['webhook'], mode='signature')
+def app_webhook(request, app: dict):
```
Comment (jefer94) [inline breathecode/authenticate/views.py:2344]: Ask

**Review by tommygonzaleza — APPROVED** [review]

**Pull request merged**